### PR TITLE
Bugfix/ie glsl preprocessor shortcoming

### PR DESCRIPTION
**Pull request by Kuranes**
- IE doesn't support preprocessor inside function parameters https://connect.microsoft.com/IE/Feedback/Details/1218138
- IE has its own way of reporting shader error => fixed.

### CODE REVIEW

File: sources/osg/Shader.js [hunk 1]
```diff
@@ -47,6 +47,12 @@ define( [
             var lines = source.split( '\n' );
             var i, m;
 
+            // IE reporting is not the same
+            if (r.exec(errors) === null){
+             r = /Shader compilation errors\n\((\d+)\, \d+\): (.+)/gmi;
```
Comment (cedricpinson) [inline sources/osg/Shader.js]: indent is not good

File: sources/osgShadow/shaders/evsm.glsl [hunk 2]
```diff
@@ -11,4 +11,22 @@ vec2 warpDepth(const in float depth, const in vec2 exponents)
...
+float computeShadow(const in bool lighted,
+                    in vec4 shadowVertexProjected,
+                    const in sampler2D tex,
+                    const in vec4 shadowMapSize,
+                    const in vec4 depthRange,
+                    const in vec3 LightPosition,
+                    const in float N_Dot_L,
+                    const in vec3 Normal,
+                    const in float bias
+                    , const in float epsilonVSM
```
Comment (cedricpinson) [inline sources/osgShadow/shaders/evsm.glsl]: coding style coherence

Comment (Kuranes) [inline sources/osgShadow/shaders/evsm.glsl]: fixed

File: sources/osg/Shader.js (hunk 1 above)
Comment (Kuranes) [inline sources/osg/Shader.js]: fixed

File: sources/osgShadow/shaders/shadowsReceive.glsl [hunk 3]
```diff
@@ -33,150 +33,4 @@ vec4 getQuadFloatFromTex(sampler2D depths, vec2 uv){
...
-    shadow = min(posContrib, negContrib);
-
-#endif
-
-
-    return shadow;
-
-}
-// end shadows
+#pragma include "PCF.glsl" "_NONE" "_PCF"
```
Comment (cedricpinson) [inline sources/osgShadow/shaders/shadowsReceive.glsl]: PCF upper case esm lower case ... move PCF to lower case

Comment (Kuranes) [inline sources/osgShadow/shaders/shadowsReceive.glsl]: fixed

**Pull request merged**